lnrpc/routerrpc/router_server.go

```diff
@@ -627,12 +628,13 @@ func (s *Server) probePaymentRequest(ctx context.Context, paymentRequest string,
 // isLSP checks if the route hints indicate an LSP. An LSP is indicated with
 // true if the last node in each route hint has the same node id, false
 // otherwise.
-func isLSP(routeHints [][]zpay32.HopHint) bool {
+func isLSP(routeHints [][]zpay32.HopHint, routerBackend *RouterBackend) bool {
 	if len(routeHints) == 0 || len(routeHints[0]) == 0 {
 		return false
 	}
 
-	refNodeID := routeHints[0][len(routeHints[0])-1].NodeID
+	refHint := routeHints[0][len(routeHints[0])-1]
```

Comment:
could we rename the `ref` prefixes to something that's more explanatory?

Reply:
renamed it to `destinationHopHint` and also referenced that in the comments.